cmd/fyne_demo/tutorials/widget.go

```diff
@@ -273,8 +273,8 @@ func makeFormTab(_ fyne.Window) fyne.CanvasObject {
 
 	form := &widget.Form{
 		Items: []*widget.FormItem{
-			{Text: "Name", Widget: name},
-			{Text: "Email", Widget: email},
+			{Text: "Name", Widget: name, HintText: "Your full name"},
+			{Text: "Email", Widget: email, HintText: "A valid email address"},
```

Comment:
Don't you think should be a great idea of having NewEntryFormField or NewEntryFormInput (just as shorthands), that encapsulate `{Text: "Name", Widget: name, HintText: "Your full name"}`, then user does not have to apparently create two objects? Something like:
```
e := widget.NewEntryFormField("Label", "HintText")
[]*widget.FormItem{
   e, ...
}
```

Reply:
Well, it is an interesting question, but I don't think the code illustration works - as it seemingly moves the fields into Entry and would require a new constructor for every type of input/widget, as the current design can apply to any widget.

Normally our constructors cover only required elements, which is why HintText would not really fit.

Reply:
> Well, it is an interesting question, but I don't think the code illustration works - as it seemingly moves the fields into Entry

Not really, there could be a file called form_inputs.go and move all formInputs related to that file and define there these new constructors, maybe??

> Normally our constructors cover only required elements, which is why HintText would not really fit.

Yes, you are right, maybe it could be:

```
e := widget.NewEntryFormField("Label").SetHintText("HintText")
[]*widget.FormItem{
   e, ...
}
```

Reply:
That style of code ("chaining") is something we discussed but have not yet decided upon.
As far as I can see your code is introducing a `FormEntry` type to avoid using the `FormItem` metadata struct.
I'm not following why this is better, sorry.

Reply:
> That style of code ("chaining") is something we discussed but have not yet decided upon.

I see, sorry. Although I think would be a nice idea that kind of style specially if the constructors only have required parameters. Or maybe create a method SetOpts(&WidgetOpts{}), so options can be unified.

> As far as I can see your code is introducing a FormEntry type to avoid using the FormItem metadata struct.

Actually, I was thinking on upgrade FormItem to an actual widget that holds hint/error texts and label.

> I'm not following why this is better, sorry. 

I think in this way, label and entry is more compact, as label is defined at the same time as the entry and not later in the form. Also this will let us to use that entry variant with hint/error text outside Form. However, it is just style decision, maybe I am wrong.

Reply:
> Actually, I was thinking on upgrade FormItem to an actual widget that holds hint/error texts and label.

That approach precludes the ability for labels to be sometimes-left and sometimes-above for different form layouts.

> Also this will let us to use that entry variant with hint/error text outside Form

This is a good point, and it came up over the last few days. I think the agreement was that we land in form, see how it works out, and if we need to extrat the behaviour to another widget we could do so later.

Reply:
> That approach precludes the ability for labels to be sometimes-left and sometimes-above for different form layouts.

That's true, however I think we can manage a private property, in order to reserve or not space for label, then if horizontal alignment is used, there is no need to reserve space for label, and it will just look as now. 

> This is a good point, and it came up over the last few days. I think the agreement was that we land in form, see how it works out, and if we need to extrat the behaviour to another widget we could do so later.

Looks good to me.

Reply:
>  in order to reserve or not space for label, 

I don't think the issue would be that it would not reserve the space so much as how do we tell this new widget not to draw the label because we want it on the left instead?

Reply:
> I don't think the issue would be that it would not reserve the space so much as how do we tell this new widget not to draw the label because we want it on the left instead?

Yes, you are right, something like that can be done. After looking at the available widgets, I wonder what other widgets (apart from Entry) are benefited with helper/error text decoration. It seems Entry is the only one, am I missing something?